runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -340,12 +359,19 @@ public String toString() {
      * Verify that max payload is enforced for the specified limit.
      *
      * @param limit Max write limit.
+     * @return the serialized size of the payload
      */
-    public void checkMaxWriteSize(int limit) {
-        try (ILogData.SerializationHandle sh = this.getSerializedForm()) {
-            if (limit != 0 && getSizeEstimate() > limit) {
-                throw new WriteSizeException(getSizeEstimate(), limit);
-            }
+    public int checkMaxWriteSize(int limit) {
+        if (serializedCache == null) {
+            throw new IllegalStateException("checkMaxWriteSize requires serialized form");
         }
+
+        int payloadSize = getSizeEstimate();
+
+        if (limit != 0 && payloadSize > limit) {
```

Comment:
Can remove the first clause if you change maxWriteSize to Integer.MAX_VALUE

Reply:
Done.